File: Golang/Easy/3005_Count_Elements_With_Maximum_Frequency/solution.go
package countelementswithmaximumfrequency

func maxFrequencyElements(nums []int) int {
	maxCount := 0
	countMap := make(map[int]int)

	for _, num := range nums {
		countMap[num] += 1
		maxCount = max(maxCount, countMap[num])
	}

	count := 0
	for num, _ := range countMap {
		if countMap[num] == maxCount {
			count += countMap[num]
		}
	}
	return count
}

File: Python/Easy/3005_Count_Elements_With_Maximum_Frequency/solution.py
class Solution:
    def maxFrequencyElements(self, nums: list[int]) -> int:
        maxCount = 0
        countMap = {}

        for num in nums:
            if num not in countMap:
                countMap[num] = 0
            countMap[num] += 1
            maxCount = max(maxCount, countMap[num])

        count = 0
        for num in countMap:
            if countMap[num] == maxCount:
                count += countMap[num]
        return count